pkg/cmd/server/api/v1/types.go

```diff
@@ -99,6 +99,11 @@ type MasterConfig struct {
 	// PauseControllers instructs the master to not automatically start controllers, but instead
 	// to wait until a notification to the server is received before launching them.
 	PauseControllers bool `json:"pauseControllers,omitempty"`
+	// ControllerLeaseTTL enables controller election, instructing the master to attempt to acquire
+	// a lease before controllers start and renewing it within a number of seconds defined by this value.
+	// Setting this value non-negative forces pauseControllers=true. This value defaults off (0, or
+	// omitted) and controller election can be disabled with -1.
+	ControllerLeaseTTL int `json:"controllerLeaseTTL,omitempty"`
```

Comment:
Can you guys complete your review by this afternoon on this?

On Mon, Aug 17, 2015 at 10:22 AM, David Eads <notifications@github.com>
wrote:

> In pkg/cmd/server/api/v1/types.go
> <https://github.com/openshift/origin/pull/4137#discussion_r37192083>:
>
> > @@ -99,6 +99,11 @@ type MasterConfig struct {
> >  	// PauseControllers instructs the master to not automatically start controllers, but instead
> >  	// to wait until a notification to the server is received before launching them.
> >  	PauseControllers bool `json:"pauseControllers,omitempty"`
> > +	// ControllerLeaseTTL enables controller election, instructing the master to attempt to acquire
> > +	// a lease before controllers start and renewing it within a number of seconds defined by this value.
> > +	// Setting this value non-negative forces pauseControllers=true. This value defaults off (0, or
> > +	// omitted) and controller election can be disabled with -1.
> > +	ControllerLeaseTTL int `json:"controllerLeaseTTL,omitempty"`
>
> It's omitempty, so it doesn't add anything. Hrm... shouldn't be omit empty.
>
> Yeah, nothing should be omitempty.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4137/files#r37192083>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift